src/core/Akka/Actor/Scheduler/HashedWheelTimerScheduler.cs

```diff
@@ -34,8 +38,45 @@ namespace Akka.Actor
     /// </summary>
     public class HashedWheelTimerScheduler : SchedulerBase, IDateTimeOffsetNowTimeProvider, IDisposable
     {
+        /// <summary>
+        /// When there are too small number of ticks to wait, use SpinWait instead
+        /// </summary>
+        private const int TicksToUseThreadSpinWait = 100;
+        
         private readonly TimeSpan _shutdownTimeout;
         private readonly long _tickDuration; // a timespan expressed as ticks
+        
+        private long _startTime = 0;
+        private long _tick;
+        private readonly int _mask;
+        private readonly Stopwatch _sleepWatch = new Stopwatch();
+        private readonly CountdownEvent _workerInitialized = new CountdownEvent(1);
+        private readonly ConcurrentQueue<SchedulerRegistration> _registrations = new ConcurrentQueue<SchedulerRegistration>();
+        private readonly Bucket[] _wheel;
+        private Thread _worker;
+
+        private readonly HashSet<SchedulerRegistration> _unprocessedRegistrations = new HashSet<SchedulerRegistration>();
+        private readonly HashSet<SchedulerRegistration> _rescheduleRegistrations = new HashSet<SchedulerRegistration>();
+
+        private const int WORKER_STATE_INIT = 0;
+        private const int WORKER_STATE_STARTED = 1;
+        private const int WORKER_STATE_SHUTDOWN = 2;
+        
+        /// <summary>
+        /// 0 - init, 1 - started, 2 - shutdown
+        /// </summary>
+        private volatile int _workerState = WORKER_STATE_INIT;
+
+        /// <summary>
+        /// API for internal usage
+        /// </summary>
+        [InternalApi]
+        public static readonly AtomicCounter TotalTicksRequiredToWaitStrict = new AtomicCounter(0);
```

Comment:
What are these being used for exactly? Testing?

Reply:
Indeed, some more comments required here... Yes, for testing - I will update summary now.